python/pyspark/sql/connect/expressions.py

```diff
@@ -414,6 +414,12 @@ def to_plan(self, session: "SparkConnectClient") -> proto.Expression:
         return fun
 
     def __repr__(self) -> str:
+        # Special handling for certain infix operators that require slightly
+        # different printing.
+        if len(self._args) == 2 and len(self._name) == 1:
```

Comment:
Actually, what about we just whitelist, and make it nicer? For example, there are only a fixed number of operators (20 ~ 30) that PySpark supports plug `eqNullSafe`, and very likely we won't add some more about this.

Otherwise, I would just even remove these new logics, and just fix the doctest with `...`.

Reply:
Let me just take it over at https://github.com/apache/spark/pull/39616 by doing a (very) simple version for now to make the tests pass.